tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedTestFormatTestRunner.cs

```diff
@@ -0,0 +1,214 @@
+                test: testCase.Test);
+        }
+
+        public void Run(
+            string schemaVersion,
+            BsonArray runOnRequirements,
+            BsonArray entities,
+            BsonArray initialData,
+            BsonDocument test)
+        {
+            schemaVersion.Should().StartWith("1.0");
+            if (runOnRequirements != null)
+            {
+                RequireServer.Check().RunOn(runOnRequirements);
+            }
+
+            _entityMap = new EntityMap(entities);
+
+            if (initialData != null)
+            {
+                AddInitialData(DriverTestConfiguration.Client, initialData);
+            }
+
+            foreach (var operationItem in test["operations"].AsBsonArray)
+            {
+                var operation = CreateOperation(operationItem.AsBsonDocument, _entityMap);
+                var result = ExecuteOperation(operation, test["async"].AsBoolean);
+                if (result != null)
+                {
+                    AssertOperation(operationItem.AsBsonDocument, result, _entityMap);
+                }
+                if (operation is UnifiedFailPointOperation failPointOperation)
+                {
+                    _failPoints.Add(failPointOperation.FailPoint);
+                }
+            }
+
+            if (test.AsBsonDocument.TryGetValue("outcome", out var expectedOutcome))
+            {
+                AssertOutcome(DriverTestConfiguration.Client, expectedOutcome.AsBsonArray);
+            }
+            if (test.AsBsonDocument.TryGetValue("expectEvents", out var expectedEvents))
+            {
+                AssertEvents(expectedEvents.AsBsonArray, _entityMap);
+            }
+        }
+
+        public void Dispose()
+        {
+            foreach (var failPoint in _failPoints)
+            {
+                failPoint?.Dispose();
+            }
+            _entityMap?.Dispose();
+        }
+
+        // private methods
+        private void AddInitialData(IMongoClient client, BsonArray initialData)
+        {
+            foreach (var dataItem in initialData)
+            {
+                var collectionName = dataItem["collectionName"].AsString;
+                var databaseName = dataItem["databaseName"].AsString;
+                var documents = dataItem["documents"].AsBsonArray.Cast<BsonDocument>().ToList();
+
+                if (documents.Count > 0)
+                {
+                    var database = client.GetDatabase(databaseName);
+                    var collection = database
+                        .GetCollection<BsonDocument>(collectionName)
+                        .WithWriteConcern(WriteConcern.WMajority);
+
+                    database.DropCollection(collectionName);
+                    collection.InsertMany(documents);
+                }
+            }
+        }
+
+        private void AssertOperation(BsonDocument operation, OperationResult operationResult, EntityMap entityMap)
```

Comment:
Let's call this `AssertResult` (because that's what we're doing):

    private void AssertResult(OperationResult actualResult, BsonDocument operation, EntityMap entityMap)

Note also renaming `operationResult` to `actualResult` and actual before expected.

(expected result is implied by and will be extracted from the operation/entityMap pair)